utils.sh

```diff
@@ -95,3 +95,7 @@ function network_ip() {
     fi
     echo "$ip"
 }
+
+function urldecode() {
+  echo -e "$(sed 's/+/ /g;s/%\(..\)/\\x\1/g;')"
```

Comment:
nit: will this work for UTF8 URLs?

Reply:
Its a temporary fix that should be removed again soon, so going to merge as it works with what we currently have in the URL.